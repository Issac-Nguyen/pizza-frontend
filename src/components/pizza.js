import React , {useState} from 'react';
import {Row, Col, Select, Button} from 'antd';
import { connect } from 'react-redux';
import {bookAction} from '../actions/orderActions'
import PropTypes from 'prop-types';

export const PizzaItem = (props) => {
    const addPizza = (e) => {
        const pizzaInfo = {
            id: props.id,
            name: props.name,
            url: props.url,
            price: props.price,
            currency: props.currency
        }
        props.book(pizzaInfo);
    }
    
    return (
        <Col span={5} className="item" id={props.id}>
            <div className="item-img">
                <img src={props.url} />
            </div>
            <div className="item-des">
                <div className="name">
                    {props.name}
                </div>
                <div className="price">
                    ${props.price}
                </div>
                <Button className="add-cart" type="danger" onClick={addPizza}>Add to cart</Button>
            </div>
        </Col>
    )
}

PizzaItem.propTypes = {
    id: PropTypes.number.isRequired,
    url: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    price: PropTypes.number.isRequired
}

  
   const mapDispatchToProps = dispatch => ({
    book: (pizza) => {dispatch(bookAction(pizza))}
   })

export default connect(undefined, mapDispatchToProps)(PizzaItem);