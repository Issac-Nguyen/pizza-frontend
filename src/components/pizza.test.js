import React from 'react'
import {render, cleanup} from '@testing-library/react'
import {PizzaItem as Pizza} from './pizza'

Object.defineProperty(window, 'matchMedia', {
    writable: true,
    value: jest.fn().mockImplementation(query => ({
      matches: false,
      media: query,
      onchange: null,
      addListener: jest.fn(), // deprecated
      removeListener: jest.fn(), // deprecated
      addEventListener: jest.fn(),
      removeEventListener: jest.fn(),
      dispatchEvent: jest.fn(),
    })),
  });

  describe('Pizza', () => {
    afterEach(cleanup);

    it('should take a snapshot', () => {
        const pizza = {id: 1, url: 'http://google.com', name:"pizza 1", price:10}
        const { asFragment } = render(<Pizza {...pizza}/>);
        
        expect(asFragment(<Pizza {...pizza}/>)).toMatchSnapshot()
     })

     it('should have name pizza 1', () => {
        const pizza = {id: 1, url: 'http://google.com', name:"pizza 1", price:10}
        const { container } = render(<Pizza {...pizza}/>);
        expect(container.querySelector('.item')).toHaveTextContent('pizza 1')
     })

})