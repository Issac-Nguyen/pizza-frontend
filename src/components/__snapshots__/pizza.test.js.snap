// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pizza should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="ant-col ant-col-5 item"
    id="1"
  >
    <div
      class="item-img"
    >
      <img
        src="http://google.com"
      />
    </div>
    <div
      class="item-des"
    >
      <div
        class="name"
      >
        pizza 1
      </div>
      <div
        class="price"
      >
        $10
      </div>
      <button
        class="ant-btn add-cart ant-btn-danger"
        type="button"
      >
        <span>
          Add to cart
        </span>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
